core/analysis/text_token_normalizing_stream.cpp

```diff
@@ -117,85 +118,80 @@ constexpr frozen::unordered_map<
 };
 
 
-bool parse_json_config(
-    const irs::string_ref& args,
+bool parse_vpack_options(
+    const VPackSlice slice,
     irs::analysis::text_token_normalizing_stream::options_t& options) {
-  rapidjson::Document json;
-  if (json.Parse(args.c_str(), args.size()).HasParseError()) {
-    IR_FRMT_ERROR(
-        "Invalid jSON arguments passed while constructing "
-        "text_token_normalizing_stream, arguments: %s",
-        args.c_str());
 
+  if (!slice.isObject() && !slice.isString()) {
+    IR_FRMT_ERROR(
+      "Slice for delimited_token_stream is not an object or string");
     return false;
   }
 
   try {
-    switch (json.GetType()) {
-      case rapidjson::kStringType:
-        return make_locale_from_name(json.GetString(), options.locale);  // required
-      case rapidjson::kObjectType:
-        if (json.HasMember(LOCALE_PARAM_NAME.c_str()) &&
-            json[LOCALE_PARAM_NAME.c_str()].IsString()) {
-          if (!make_locale_from_name(json[LOCALE_PARAM_NAME.c_str()].GetString(), options.locale)) {
+    switch (slice.type()) {
+      case VPackValueType::String:
+        return make_locale_from_name(irs::get_string<irs::string_ref>(slice), options.locale);  // required
+      case VPackValueType::Object:
+        if (slice.hasKey(LOCALE_PARAM_NAME) &&
```

Comment:
Let`s get LOCALE_PARAM_NAME slice only once. And than we could check if it  `isNone` or `isString` or anything else (which is an error here)